tests/components/onewire/test_entity_sysbus.py

```diff
@@ -0,0 +1,92 @@
+"""Tests for 1-Wire temperature sensor (device family 10, 22, 28, 3B, 42) connected on SysBus."""
+from os import path
+from unittest.mock import PropertyMock, patch
+
+from pi1wire import (
+    InvalidCRCException,
+    NotFoundSensorException,
+    UnsupportResponseException,
+)
+
+from homeassistant.components.onewire.const import DEFAULT_SYSBUS_MOUNT_DIR, DOMAIN
+from homeassistant.components.onewire.sensor import OneWireDirect
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.setup import async_setup_component
+from homeassistant.util import slugify
+
+from tests.common import mock_registry
+
+MOCK_DEVICE_ID = "28-111111111111"
+MOCK_DEVICE_NAME = "My DS18B20"
+
+
+async def test_onewiredirect_setup(hass):
+    """Test that sysbus config entry works correctly."""
+    entity_registry = mock_registry(hass)
+    config = {
+        "sensor": {
+            "platform": DOMAIN,
+            "names": {
+                MOCK_DEVICE_ID: MOCK_DEVICE_NAME,
+            },
+        }
+    }
+    with patch(
+        "homeassistant.components.onewire.sensor.Pi1Wire"
+    ) as mock_pi1wire, patch("pi1wire.OneWire") as mock_owsensor:
+        type(mock_owsensor).mac_address = PropertyMock(
+            return_value=MOCK_DEVICE_ID.replace("-", "")
+        )
+        mock_owsensor.get_temperature.return_value = "25.123"
+        mock_pi1wire.return_value.find_all_sensors.return_value = [mock_owsensor]
+        assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+        await hass.async_block_till_done()
+
+        assert len(entity_registry.entities) == 1
+        sensor_id = "sensor." + slugify(MOCK_DEVICE_NAME) + "_temperature"
+        sensor_entity = entity_registry.entities.get(sensor_id)
```

Comment:
Please rename `sensor_entity` to `sensor_entry` or similar. This is not an Entity instance.